Add migration phases pages (#8828)
opensearch-project/documentation-website: _migrations/migration-console/migration-console-commands-references.md

---
layout: default
title: Command reference
nav_order: 35
parent: Migration console
---

# Migration console command reference

Migration console commands follow this syntax: `console [component] [action]`. The components include `clusters`, `backfill`, `snapshot`, `metadata`, and `replay`. The console is configured with a registry of the deployed services and the source and target cluster, generated from the `cdk.context.json` values.

## Commonly used commands

The exact commands used will depend heavily on use-case and goals, but the following are a series of common commands with a quick description of what they do.

### Check connection

Reports whether both the source and target clusters can be reached and their versions.

> 

Author: Yes. I'll update this.

```sh
console clusters connection-check
```

### Run `cat-indices`

> Runs the Cat Indices API on the cluster.

Author: ```suggestion
Runs the `cat-indices` API on the cluster.
```

```sh
console clusters cat-indices
```

### Create a snapshot

> Creates a snapshot on the source cluster, into a pre-configured Amazon S3 bucket.

Author: ```suggestion
Creates a snapshot on the source cluster, into a preconfigured Amazon S3 bucket.
```

```sh
console snapshot create
```

## Check snapshot status

Runs a detailed check on the status of the snapshot creation, including estimated completion time.

```sh
console snapshot status --deep-check
```

## Evaluate metadata

Performs a dry run of metadata migration, showing which indexes, templates, and other objects will be migrated to the target cluster.

```sh
console metadata evaluate
```

## Migrate metadata

Migrates the metadata from the source cluster to the target cluster.

```sh
console metadata migrate
```

## Start a backfill

If the `Reindex-From-Snapshot` (RFS) is enabled, starts an instance of the service to begin moving documents to the target cluster.

There are similar `scale UNITS` and `stop` commands to change the number of active instances for RFS.

```sh
console backfill start
```

> ## Chack backfill status

Author: ```suggestion
## Check backfill status
```

Gets the current status of the backfill migration, with the number of instances operating and the progress of the shards.

## Start Traffic Replayer

If the Traffic Replayer is enabled, starts an instance of the service to begin replaying traffic against the target cluster.
The `stop` command stops all active instances.

```sh
console replay start
```

## Read logs

> Reads any logs that exist when running Traffic Replayer.Use tab completion on the path to fill in the available `NODE_IDs` and, if applicable, log file names. The tuple's logs roll over at a certain size threshold, so there may be many files named with timestamps. The `jq` command pretty-prints each line of the tuple output before writing it to file.

Author: ```suggestion
Reads any logs that exist when running Traffic Replayer. Use tab completion on the path to fill in the available `NODE_IDs` and, if applicable, log file names. The tuples logs roll over at a certain size threshold, so there may be many files named with timestamps. The `jq` command pretty-prints each line of the tuple output before writing it to file.
```

```sh
console tuples show --in /shared-logs-output/traffic-replayer-default/[NODE_ID]/tuples/console.log | jq > readable_tuples.json
```

## Help command

All commands and options can be explored within the tool itself by using the `--help` option, either for the entire `console` application or for individual components (e.g. `console backfill --help`). The console also has command autocomplete set up to assist with usage. For example:

> 

Author: Sentence is unnecessary, so removing it.
